bson/json_util.py

```diff
@@ -499,4 +587,19 @@ def default(obj, json_options=DEFAULT_JSON_OPTIONS):
             return {"$uuid": obj.hex}
     if isinstance(obj, Decimal128):
         return {"$numberDecimal": str(obj)}
+    if isinstance(obj, bool):
+        return obj
+    if json_options.canonical_extended_json and isinstance(obj, integer_types):
+        return {'$numberInt': text_type(obj)}
```

Comment:
This will not produce canonical JSON when the int is outside the bounds of a 32-bit integer. It should produce '$numberInt', '$numberLong', or error when its outside 64 bits.